src/dmd/target.d

```diff
@@ -1306,12 +1306,12 @@ struct TargetCPP
      */
     extern (C++) const(char)* typeInfoMangle(ClassDeclaration cd)
     {
-        static if (TARGET.Linux || TARGET.OSX || TARGET.FreeBSD || TARGET.OpenBSD || TARGET.Solaris || TARGET.DragonFlyBSD)
+        if (target.os & (Target.OS.linux | Target.OS.OSX | Target.OS.FreeBSD | Target.OS.OpenBSD | Target.OS.Solaris | Target.OS.DragonFlyBSD))
             return cppTypeInfoMangleItanium(cd);
-        else static if (TARGET.Windows)
+        if (target.os == Target.OS.Windows)
```

Comment:
this does make `target.cpp` dependant on `target`. I'm not sure what the best way to do this is. Its still an improvement on the status quo though.

Reply:
I guess `TARGET` could instead be made private to `dmd.target` first, then deal with how you want to be able to support cross-compilation later.

I assume you want to have this as a compiler flag, so there should be no static conditions anywhere for target decisions.

Reply:
> I guess TARGET could instead be made private to dmd.target first, then deal with how you want to be able to support cross-compilation later.

This is not a problem provided that `target` remains a global singleton.

> I assume you want to have this as a compiler flag, so there should be no static conditions anywhere for target decisions.

Correct.